openpifpaf/show/painters.py

```diff
@@ -18,14 +18,14 @@
 class AnnotationPainter:
     def __init__(self, *,
                  xy_scale=1.0,
-                 keypoint_painter=None,
-                 crowd_painter=None,
-                 detection_painter=None):
-        self.painters = {
-            'Annotation': keypoint_painter or KeypointPainter(xy_scale=xy_scale),
-            'AnnotationCrowd': crowd_painter or CrowdPainter(xy_scale=xy_scale),
-            'AnnotationDet': detection_painter or DetectionPainter(xy_scale=xy_scale),
-        }
+                 painters=None):
+        self.painters = {}
+        for painter_key, painter in PAINTERS.items():
+            if painters:
+                self.painters[painter_key] = painters.get(painter_key, None) or\
+                                                painter(xy_scale=xy_scale)
```

Comment:
You have two instances of painter here. You can avoid this by testing for `'annotation_type' in painters`.